pvnet_app/utils.py

```diff
@@ -66,18 +64,58 @@ def populate_data_config_sources(input_path, output_path, gsp_path=""):
     if "nwp" in config["input_data"]:
         nwp_config = config["input_data"]["nwp"]
         for nwp_source in nwp_config.keys():
-            if nwp_config[nwp_source][f"nwp_zarr_path"]!="":
-                assert "nwp" in production_paths, f"Missing production path: nwp"
+            if nwp_config[nwp_source]["nwp_zarr_path"]!="":
+                assert "nwp" in production_paths, "Missing production path: nwp"
                 assert nwp_source in production_paths["nwp"], f"Missing NWP path: {nwp_source}"
-                nwp_config[nwp_source][f"nwp_zarr_path"] = production_paths["nwp"][nwp_source]
+                nwp_config[nwp_source]["nwp_zarr_path"] = production_paths["nwp"][nwp_source]
+                
+    return config
+
+
+def overwrite_config_dropouts(config: dict) -> dict:
+    """Overwrite the config drouput parameters for production
+
+    Args:
+        config: The data config
+    """
+    
+    # Replace data sources
+    for source in ["satellite"]:
+        if source in config["input_data"] :
+            if config["input_data"][source][f"{source}_zarr_path"]!="":
+                config["input_data"][source][f"dropout_timedeltas_minutes"] = None
+        
+    # NWP is nested so much be treated separately
+    if "nwp" in config["input_data"]:
+        nwp_config = config["input_data"]["nwp"]
+        for nwp_source in nwp_config.keys():
+            if nwp_config[nwp_source]["nwp_zarr_path"]!="":
+                nwp_config[nwp_source]["dropout_timedeltas_minutes"] = None
+                
+    return config
+    
+    
+def modify_data_config_for_production(
+    input_path: str, 
+    output_path: str, 
+    gsp_path: str = ""
+) -> None:
+    """Resave the data config with the data source filepaths and dropouts overwritten
 
-    # We do not need to set PV path right now. This currently done through datapipes
-    # TODO - Move the PV path to here
+    Args:
+        input_path: Path to input datapipes configuration file
+        output_path: Location to save the output configuration file
+        gsp_path: For lagacy usage only
+    """
+    config = load_yaml_config(input_path)
+    
+    config = populate_config_with_data_data_filepaths(config, gsp_path=gsp_path)
+    config = overwrite_config_dropouts(config)
```

Comment:
this is very clear and nice

Reply:
Thanks. Although I've just noticed the double data in the name `populate_config_with_data_data_filepaths()`. I've just merged so I'll leave this one for the refactor